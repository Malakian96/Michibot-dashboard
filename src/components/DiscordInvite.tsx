import React from 'react';
import Image from 'next/image';

const DiscordInvite = () => {
  const inviteLink =
    'https://discord.com/oauth2/authorize?client_id=1303130781580918954&permissions=8&integration_type=0&scope=bot';

  return (
    <div className="bg-discord-dark-light text-foreground max-w-md mx-auto my-10 p-6 rounded-lg shadow-lg">
      <div className="flex justify-center pb-5">
        <Image
          src={'/thebiter.png'}
          alt="User avatar"
          width={100}
          height={100}
          className="rounded-full"
        />
      </div>
      <h2 className="text-2xl font-bold text-white flex items-center justify-center">
        Michibot
        <span className="ml-2 bg-discord-blurple text-white text-xs font-semibold px-2 py-1 rounded">
          APP
        </span>
      </h2>
      <p className="text-discord-gray mt-2 text-center">
        Añade a Michibot a tu servidor de Discord y accede a todas sus
        funcionalidades.
      </p>

      {/* Lista de permisos */}
      <ul className="mt-4 space-y-2 text-sm text-foreground">
        <li className="flex items-center">
          ✅ <span className="ml-2">Añadir un bot a un servidor</span>
        </li>
        <li className="flex items-center">
          ✅ <span className="ml-2">Crear comandos</span>
        </li>
        <li className="flex items-center">
          ❌{' '}
          <span className="ml-2 text-discord-gray">
            Tener una crisis existencial
          </span>
        </li>
      </ul>

      <div className="mt-6 text-center">
        <a
          href={inviteLink}
          target="_blank"
          rel="noopener noreferrer"
          className="bg-discord-blurple hover:bg-discord-blurple/90 text-white font-semibold py-2 px-4 rounded shadow-md transition duration-300"
        >
          Invitar al servidor
        </a>
      </div>
    </div>
  );
};

export default DiscordInvite;
