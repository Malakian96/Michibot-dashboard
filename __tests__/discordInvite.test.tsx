import React from 'react';
import { render } from '@testing-library/react';
import DiscordInvite from '../src/components/DiscordInvite';

describe('DiscordInvite', () => {
  it('renders the heading correctly', () => {
    const { getByText } = render(<DiscordInvite />);
    expect(getByText('Michibot')).toBeInTheDocument();
  });

  it('renders the description correctly', () => {
    const { getByText } = render(<DiscordInvite />);
    expect(
      getByText(
        'Añade a Michibot a tu servidor de Discord y accede a todas sus funcionalidades.',
      ),
    ).toBeInTheDocument();
  });

  it('renders the permissions list correctly', () => {
    const { getByText } = render(<DiscordInvite />);
    expect(getByText('Añadir un bot a un servidor')).toBeInTheDocument();
    expect(getByText('Crear comandos')).toBeInTheDocument();
    expect(getByText('Tener una crisis existencial')).toBeInTheDocument();
  });

  it('renders the invite link correctly', () => {
    const inviteLink =
      'https://discord.com/oauth2/authorize?client_id=1303130781580918954&permissions=8&integration_type=0&scope=bot';
    const { getByRole } = render(<DiscordInvite />);
    expect(getByRole('link')).toHaveAttribute('href', inviteLink);
  });
});

describe('DiscordInviteSnapshot', () => {
  it('matches snapshot', () => {
    const { container } = render(<DiscordInvite />);
    expect(container).toMatchSnapshot();
  });
});
