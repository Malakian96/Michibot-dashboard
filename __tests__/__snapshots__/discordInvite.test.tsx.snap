// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscordInviteSnapshot matches snapshot 1`] = `
<div>
  <div
    class="bg-discord-dark-light text-foreground max-w-md mx-auto my-10 p-6 rounded-lg shadow-lg"
  >
    <div
      class="flex justify-center pb-5"
    >
      <img
        alt="User avatar"
        class="rounded-full"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/_next/image?url=%2Fthebiter.png&w=256&q=75"
        srcset="/_next/image?url=%2Fthebiter.png&w=128&q=75 1x, /_next/image?url=%2Fthebiter.png&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
    </div>
    <h2
      class="text-2xl font-bold text-white flex items-center justify-center"
    >
      Michibot
      <span
        class="ml-2 bg-discord-blurple text-white text-xs font-semibold px-2 py-1 rounded"
      >
        APP
      </span>
    </h2>
    <p
      class="text-discord-gray mt-2 text-center"
    >
      Añade a Michibot a tu servidor de Discord y accede a todas sus funcionalidades.
    </p>
    <ul
      class="mt-4 space-y-2 text-sm text-foreground"
    >
      <li
        class="flex items-center"
      >
        ✅ 
        <span
          class="ml-2"
        >
          Añadir un bot a un servidor
        </span>
      </li>
      <li
        class="flex items-center"
      >
        ✅ 
        <span
          class="ml-2"
        >
          Crear comandos
        </span>
      </li>
      <li
        class="flex items-center"
      >
        ❌
         
        <span
          class="ml-2 text-discord-gray"
        >
          Tener una crisis existencial
        </span>
      </li>
    </ul>
    <div
      class="mt-6 text-center"
    >
      <a
        class="bg-discord-blurple hover:bg-discord-blurple/90 text-white font-semibold py-2 px-4 rounded shadow-md transition duration-300"
        href="https://discord.com/oauth2/authorize?client_id=1303130781580918954&permissions=8&integration_type=0&scope=bot"
        rel="noopener noreferrer"
        target="_blank"
      >
        Invitar al servidor
      </a>
    </div>
  </div>
</div>
`;
